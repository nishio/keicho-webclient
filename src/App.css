.App {
  text-align: left;
}

textarea {
  width: calc(100% - 10px);
  margin: 5px;
  box-sizing: border-box;
  font-size: 20px;
  border-radius: 17px;
  padding: 0 10px;
  word-break: break-all;
}
