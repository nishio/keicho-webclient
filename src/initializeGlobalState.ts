import { setGlobal } from "reactn";
import { INITIAL_LOGS, PRESET_LOGS, USE_PRESET } from "./PRESET_MESSAGES";

const INITIAL_GLOBAL_STATE = {
  TalkID: "",
  logs: USE_PRESET ? PRESET_LOGS : INITIAL_LOGS,
  lastKeywords: [] as string[],
  otherKeywords: [] as string[],
  canInput: true,
  previousTalkID: "",
};

export const initializeGlobalState = () => {
  setGlobal(INITIAL_GLOBAL_STATE);
};

type TYPE_GLOBAL_STATE = typeof INITIAL_GLOBAL_STATE;

declare module "reactn/default" {
  export interface State extends TYPE_GLOBAL_STATE {}
}
