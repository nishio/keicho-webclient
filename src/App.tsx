import React, { useEffect } from "react";
import "./App.css";
import db from "./FirestoreIO";
import { NewTalk } from "./NewTalk";

const USE_LOCAL_SERVER = false;
const IS_PRODUCTION = process.env.NODE_ENV === "production";
export const APIROOT =
  IS_PRODUCTION || !USE_LOCAL_SERVER
    ? "https://keicho.herokuapp.com/api/"
    : "http://localhost:5000/api/";

export const USE_PRESET = false;
export const INITIAL_LOGS = [
  { text: "あなたはこの会話で何が起きて欲しいですか？", user: false },
];

const ShowLog = () => {
  useEffect(() => {
    const key = "I9IlVW6Xt58NlNHAjdfq";
    console.log("reading", key);
    db.collection("yagokoro_env")
      .doc(key)
      .get()
      .then((doc: any) => {
        if (doc.exists) {
          const data = doc.data();
          console.log(data);
          const x = JSON.parse(data.json);
          console.log(x);
        } else {
          // doc.data() will be undefined in this case
          console.log("No such document!");
        }
      });
  }, []);
  return <br />;
};

function App() {
  const hash = new URLSearchParams(window.location.hash.substring(1));
  if (hash.has("talk")) {
    return <ShowLog />;
  } else {
    return <NewTalk />;
  }
}

export default App;
