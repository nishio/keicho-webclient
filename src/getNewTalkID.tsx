import { ERROR_ON_SERVER } from "./PRESET_MESSAGES";
import { setGlobal } from "reactn";
import { APIROOT } from "./App";
import { localDB } from "./localDB";
import * as Sentry from "@sentry/browser";

export const getNewTalkID = () => {
  const transaction = Sentry.startTransaction({ name: "getNewTalkID" });
  const span = transaction.startChild({ op: "getNewTalkID" });
  fetch(APIROOT + "web/create/", {
    mode: "cors",
    method: "GET",
  })
    .then((response) => {
      return response.text();
    })
    .then((text) => {
      _gotNewTalkID(text).then(() => {
        Sentry.setContext("Info", { TalkID: text });
        span.finish();
        transaction.finish();
      });
    })
    .catch(() => {
      Sentry.captureMessage("ERROR_ON_SERVER: getNewTalkID");
      setGlobal({ logs: [ERROR_ON_SERVER], canInput: false });
    });
};

// exported for test
export const _gotNewTalkID = (text: string) => {
  return localDB.talks
    .orderBy("id")
    .reverse()
    .limit(1)
    .toArray()
    .then((x) => {
      const previousTalkID = x[0]?.TalkID;
      if (previousTalkID !== undefined) {
        setGlobal({ TalkID: text, previousTalkID: x[0].TalkID });
      } else {
        setGlobal({ TalkID: text, previousTalkID: "" });
      }
      localDB.talks.add({ TalkID: text });
    });
};
