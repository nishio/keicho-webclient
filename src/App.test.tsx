import React from "react";
import App from "./App";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { initializeGlobalState } from "./initializeGlobalState";
import pretty from "pretty";

import * as getNewTalkID from "./getNewTalkID";
jest.spyOn(getNewTalkID, "getNewTalkID").mockImplementation(() => {
  getNewTalkID._gotNewTalkID("test");
});

let container: HTMLDivElement;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
  initializeGlobalState();
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
});

test("render", () => {
  act(() => {
    render(<App />, container);
  });
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
