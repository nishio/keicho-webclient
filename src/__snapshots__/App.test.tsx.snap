// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
"<div class=\\"App\\">
  <div class=\\"makeStyles-root-1\\">
    <header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4\\">
      <div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\">
        <h6 class=\\"MuiTypography-root makeStyles-title-2 MuiTypography-h6\\">Keicho</h6><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"menu\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
      </div>
    </header>
  </div>
  <div class=\\"line__container\\">
    <div class=\\"line__contents\\">
      <div class=\\"line__left\\">
        <figure><img alt=\\"nisbot\\" src=\\"icons8-bot-32.png\\"></figure>
        <div class=\\"line__left-text\\">
          <div class=\\"text\\">あなたはこの会話で何が起きて欲しいですか？</div>
        </div>
      </div>
    </div>
  </div><textarea rows=\\"1\\" aria-label=\\"textarea\\" placeholder=\\"\\" id=\\"textarea\\" style=\\"height: -4px; overflow: hidden;\\"></textarea><textarea aria-hidden=\\"true\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0);\\"></textarea><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\">🙁</span><span class=\\"MuiTouchRipple-root\\"></span></button>
  <hr id=\\"bottom\\">
</div>"
`;
